.circleci/config.yml

```diff
@@ -60,36 +59,31 @@ commands:
           name: Building App
           no_output_timeout: 30m
           command: yarn build-travis
-  compress:
-    description: "Compress"
+  deploy:
+    description: "Deploy"
+    parameters:
+      target_branch:
+        type: string
     steps:
+      - checkout
+      - attach_workspace:
+          at: www
       - run:
-          name: "Compress"
+          name: Tag build
+          command: echo "<< parameters.target_branch >> $(date)" > www/version
+      - run:
+          name: Install and configure dependencies
           command: |
-            tar -cvf artifact.tar index.html nojs-smartcharts.css dist/*
+            sudo npm install -g gh-pages@2.0.1
+            git config user.email "mahboub.meng@gmail.com"
+            git config user.name "Mahboobeh-binary"
+      - add_ssh_keys:
```

Comment:
since one git user account is being used to push changes to `gh-pages` branch hence i would suggest the workflow in the following way -> add-ssh-fingerprint-of-machine-user, git-config-user-email-and-name, verify-if-user-is-able-to-ssh-to-github.com (`ssh -Tv git@github.com`), tag-build, deploy-to-branch

Reply:
even i would suggest using git machine user's access control checking to github.com even before the build starts

Reply:
@shudarshon-binary yes that how we use it, we use machine user